shiny/_main.py

```diff
@@ -602,22 +605,18 @@ def create(
     template: Optional[str] = None,
     mode: Optional[str] = None,
     github: Optional[str] = None,
-    dir: Optional[str | Path] = None,
+    dir: Optional[Path] = None,
```

Comment:
Are we sure this is always a `Path`? If so, why keep the `str` runtime check?

Reply:
Oops, I didn't mean to remove `str` from the input typing. I do think we could use

```
type=click.Path(dir_okay=True, file_okay=False)
```

for the `--dir` option. This does guarantee that `dir` is then a `Path()`. I'd still be on the fence about allowing `str`, wdyt?

Reply:
Ok, yea, that seems sensible, and I don't see any harm in keeping the `str` type.